smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
+    def __init__(self, loss, learning_rate=0.0001, b1=0.9, b2=0.999, epsilon=1e-8, lmbda=(1 - 1e-8)):
+        super().__init__(loss)
+        self.lr = learning_rate
+        self.b1 = b1
+        self.b2 = b2
+        self.eps = epsilon
+        self.lmbda = lmbda
+
+        self._updates = OrderedDict()
+        self.t = sharedX(0, name='t')
+        self.mts = []
+        self.vts = []
+
+        for i, param in enumerate(loss.gradients):  # iterates on the parameters key
+            self.mts.append(sharedX(np.zeros_like(param.get_value()), name='mt{}'.format(i)))
```

Comment:
Where? In sharedX, we use `None` quite clearly.

On 18/09/15 12:59 PM, Marc-Alexandre Côté wrote:
>
> In smartlearner/optimizers/adam.py 
> <https://github.com/SMART-Lab/smartlearner/pull/51#discussion_r39876987>:
>
> > +class Adam(Optimizer):
> > +    def __init__(self, loss, learning_rate=0.0001, b1=0.9, b2=0.999, epsilon=1e-8, lmbda=(1 - 1e-8)):
> > +        super().__init__(loss)
> > +        self.lr = learning_rate
> > +        self.b1 = b1
> > +        self.b2 = b2
> > +        self.eps = epsilon
> > +        self.lmbda = lmbda
> > +
> > +        self._updates = OrderedDict()
> > +        self.t = sharedX(0, name='t')
> > +        self.mts = []
> > +        self.vts = []
> > +
> > +        for i, param in enumerate(loss.gradients):  # iterates on the parameters key
> > +            self.mts.append(sharedX(np.zeros_like(param.get_value()), name='mt{}'.format(i)))
>
> Shouldn't be the case since we already make sure 
> |my_shared_variable.name| is either the name or auto_name. That said, 
> it is true that's imply the use of our |sharedX| function :S.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/SMART-Lab/smartlearner/pull/51/files#r39876987>.Web Bug 
> from 
> https://github.com/notifications/beacon/AEJelHsPZJFmieB9kdsdJce_dlUItR0Dks5ozDp9gaJpZM4F_bBX.gif
>